src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -7,89 +7,357 @@
+        {
+            try
+            {
+                value = GetDouble();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetSByte(out sbyte value) { throw null; }
+        public bool TryGetSByte(out sbyte value)
+        {
+            try
+            {
+                value = GetSByte();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetUInt16(out ushort value) { throw null; }
+        public bool TryGetUInt16(out ushort value)
+        {
+            try
+            {
+                value = GetUInt16();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetUInt32(out uint value) { throw null; }
+        public bool TryGetUInt32(out uint value)
+        {
+            try
+            {
+                value = GetUInt32();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
         [CLSCompliant(false)]
-        public bool TryGetUInt64(out ulong value) { throw null; }
+        public bool TryGetUInt64(out ulong value)
+        {
+            try
+            {
+                value = GetUInt64();
+                return true;
+            }
+            catch
+            {
+                value = 0;
+                return false;
+            }
+        }
+
+        public void SetString(string value)
+        {
+            JsonWriterHelper.ValidateNumber(Encoding.UTF8.GetBytes(value).AsSpan());
+            _value = value;
+        }
 
-        public void SetString(string value) { }
-        public void SetByte(byte value) { }
-        public void SetInt32(int value) { }
-        public void SetInt16(short value) { }
-        public void SetInt64(long value) { }
-        public void SetSingle(float value) { }
-        public void SetDouble(double value) { }
+        public void SetByte(byte value)
+        {
+            _value = value.ToString();
+        }
+        public void SetInt16(short value)
+        {
+            _value = value.ToString();
+        }
+        public void SetInt32(int value)
+        {
+            _value = value.ToString();
+        }
+        public void SetInt64(long value)
+        {
+            _value = value.ToString();
+        }
+        public void SetSingle(float value)
+        {
+            _value = value.ToString();
+        }
+        public void SetDouble(double value)
+        {
+            _value = value.ToString();
```

Comment:
This probably wants a specific format (like G15).  Whatever Utf8JsonWriter does.

Reply:
Ok, I checked it - "G15" is default for ToString and Utf8JsonWriter uses default StandardFormat - that is also "G", but with unspecified precision. Should I leave the default format or change it to "G17" or something else?

Reply:
I thought we had changed Utf8JsonWriter to specify a precision.  If not, nothing is needed here :smile:.